src/main/routes/resource-applier/apply-resource-route.injectable.ts

```diff
@@ -5,17 +5,19 @@
 import { getRouteInjectable } from "../../router/router.injectable";
 import { apiPrefix } from "../../../common/vars";
 import { ResourceApplier } from "../../resource-applier";
-import { clusterRoute } from "../../router/route";
+import { payloadValidatedClusterRoute } from "../../router/route";
+import Joi from "joi";
 
-const applyResourceRouteInjectable = getRouteInjectable({
+const createResourceRouteInjectable = getRouteInjectable({
   id: "apply-resource-route",
 
-  instantiate: () => clusterRoute({
+  instantiate: () => payloadValidatedClusterRoute({
     method: "post",
     path: `${apiPrefix}/stack`,
+    payloadValidator: Joi.string(),
```

Comment:
This validator is an added functionality?

Reply:
More like clarifying functionality. The `create-resource-route` really only wants to be used with the create resource dock tabs (but is currently erroneously also used by `KubeObject.update` which is marked as deprecated currently). And since it will only be used for create resources dock tab (the edit resource dock tab uses patch resource instead as it is more robust) there is no need for it accept anything other than manifests (so strings)

Reply:
It's pretty hard for me to be sure about this. Is the change something that should be accompanied with an explanatory test for the requirement? Seems to not belong in this PR anyway :/

Reply:
I can change it with a cast if you would like, or do you mean the change of how the route is interpreted? It is a POST after all.